pyanaconda/modules/network/nm_client.py

```diff
@@ -0,0 +1,667 @@
+            s_wired.props.mac_address = bound_mac
+            log.debug("add connection: mac %s is bound to name %s",
+                      bound_mac, device_name)
+            con.add_setting(s_con)
+        else:
+            con.add_setting(s_con)
+            bind_connection(nm_client, con, network_data.bindto, device_name)
+
+        # Add s390 settings
+        if is_s390():
+            s390cfg = get_s390_settings(device_name)
+            if s390cfg['SUBCHANNELS']:
+                subchannels = s390cfg['SUBCHANNELS'].split(",")
+                s_wired.props.s390_subchannels = subchannels
+            if s390cfg['NETTYPE']:
+                s_wired.props.s390_nettype = s390cfg['NETTYPE']
+            if s390cfg['OPTIONS']:
+                opts = s390cfg['OPTIONS'].split(" ")
+                opts_dict = {k: v for k, v in (o.split("=") for o in opts)}
+                s_wired.props.s90_options = opts_dict
+
+    added_connections.insert(0, (con, device_to_activate))
+
+    for con, device_name in added_connections:
+        log.debug("add connection: %s for %s\n%s", con_uuid, device_name,
+                  con.to_dbus(NM.ConnectionSerializationFlags.ALL))
+        device_to_activate = device_name if activate else None
+        nm_client.add_connection_async(con, True, None,
+                                       _connection_added_cb,
+                                       device_to_activate)
+
+    return added_connections
+
+
+def _connection_added_cb(client, result, device_to_activate=None):
+    """Finish asynchronous adding of a connection and activate eventually.
+
+    :param device_to_activate: name of the device to be activated with the
+                                added connection.
+    :type device_to_activate: str
+    """
+    con = client.add_connection_finish(result)
+    log.debug("connection %s added:\n%s", con.get_uuid(),
+              con.to_dbus(NM.ConnectionSerializationFlags.ALL))
+    if device_to_activate:
+        device = client.get_device_by_iface(device_to_activate)
+        if device:
+            log.debug("activating with device %s", device.get_iface())
+        else:
+            log.debug("activating without device specified (not found)")
+        client.activate_connection_async(con, device, None, None)
+
+
+def create_slave_connection(slave_type, slave_idx, slave, master, settings=None):
+    """Create a slave NM connection for virtual connection (bond, team, bridge).
+
+    :param slave_type: type of slave ("bond", "team", "bridge")
+    :type slave_type: str
+    :param slave_idx: index of the slave for naming
+    :type slave_idx: int
+    :param slave: slave's device name
+    :type slave: str
+    :param master: slave's master device name
+    :type master: str
+    :param settings: list of other settings to be added to the connection
+    :type settings: list(NM.Setting)
+
+    :return: created connection
+    :rtype: NM.SimpleConnection
+    """
+
```

Comment:
Please remove the blank line.

Reply:
https://github.com/rhinstaller/anaconda/pull/1771/commits/ba752145484a43443c54981bbcfad4dc36696a16